tests/test_tutorial/test_parameter_types/test_pydantic_types/test_tutorial001.py

```diff
@@ -0,0 +1,38 @@
+import subprocess
+import sys
+
+import typer
+from typer.testing import CliRunner
+
+from docs_src.parameter_types.pydantic_types import tutorial001 as mod
```

Comment:
We need to make sure to skip these tests when `pydantic` and/or `email-validator` aren't installed (instead of running them and getting import errors and failing tests)

Reply:
Actually I made necessary changes to have everything installed for tests, I don't get the point of skipping the tests, except if we want to ensure that typer still works without pydantic, but running both flavours would be cumbersome

Reply:
I personally think it's a valid use-case for someone to be developing on Typer and to be running the test suite locally, without wanting to install `pydantic` for Pydantic-related tests if they don't care about that...

Reply:
In my experience, it's better to not skip tests on such conditions because it's "easy" to silently break `pydantic` support because someone has removed the dependency in `requirements-test.txt` and CI didn't complain.

If a developer tests in local and doesn't have `pydantic` installed, I wouldn't expect them to run the full suite including tests on `pydantic` support (though, it's a best practice to run them, with the appropriate dependencies that the repository defines)